src/Form/WorkbenchAccessByRoleForm.php

```diff
@@ -110,15 +110,17 @@ public function buildForm(array $form, FormStateInterface $form_state, AccessSch
   public function submitForm(array &$form, FormStateInterface $form_state) {
     $roles = $form_state->getValue('roles');
     $existing_roles = $form_state->getValue('existing_roles');
+    // @TODO: validate the access scheme?
+    $access_scheme = $form_state->getValue('access_scheme');
     $id = $form_state->getValue('section_id');
     foreach ($roles as $role_id => $value) {
-      // Add user to section.
+      // Add role to section.
       if ($value && !isset($existing_roles[$role_id])) {
-        $this->roleSectionStorage->addRole($this->scheme, $role_id, [$id]);
+        $this->roleSectionStorage->addRole($access_scheme, $role_id, [$id]);
       }
-      // Remove user from section.
+      // Remove role from section.
       if (!$value && isset($existing_roles[$role_id])) {
-        $this->roleSectionStorage->removeRole($this->scheme, $role_id, [$id]);
+        $this->roleSectionStorage->removeRole($access_scheme, $role_id, [$id]);
```

Comment:
Why are these changes needed?

Reply:
I'm skeptical of the use of $this->scheme inside the class instead of passing actual value through the form.